src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -0,0 +1,100 @@
+        }
+
+        public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRequestParams requestParams)
+        {
+            ConnectionInfo connInfo = this.GetConnectionInfo(requestParams.ConnectionUri);
+            ServerConnection serverConnection = ConnectionService.OpenServerConnection(connInfo, ObjectManagementService.ApplicationName);
+
+            using (var context = new ServerViewContext(requestParams, serverConnection))
+            {
+                var serverSmo = CreateServerSmoObject(context).Result;
+                if(serverSmo != null)
+                {
+                    this.serverViewInfo.ObjectInfo = new ServerInfo()
+                    {
+                        Name = serverSmo.Name,
+                        HardwareGeneration = serverSmo.HardwareGeneration,
+                        Language = serverSmo.Language,
+                        MemoryInMB = serverSmo.PhysicalMemory,
+                        OperatingSystem = serverSmo.HostDistribution,
+                        Platform = serverSmo.HostPlatform,
+                        Processors = serverSmo.Processors,
+                        IsClustered = serverSmo.IsClustered,
+                        IsHadrEnabled = serverSmo.IsHadrEnabled,
+                        IsPolyBaseInstalled = serverSmo.IsPolyBaseInstalled,
+                        IsXTPSupported = serverSmo.IsXTPSupported,
+                        Product = serverSmo.Product,
+                        ReservedStorageSizeMB = serverSmo.ReservedStorageSizeMB,
+                        RootDirectory = serverSmo.RootDirectory,
+                        ServerCollation = serverSmo.Collation,
+                        ServiceTier = serverSmo.ServiceTier,
+                        StorageSpaceUsageInGB = (int)ByteConverter.ConvertMbtoGb(serverSmo.UsedStorageSizeMB),
+                        Version = serverSmo.Version.ToString(),
+                        MinServerMemory = configService.GetConfigByName(this.server, "min server memory (MB)").ConfigValue,
+                        MaxServerMemory = configService.GetConfigByName(this.server, "max server memory (MB)").ConfigValue
+                    };
+                }
+
+                return Task.FromResult(new InitializeViewResult { ViewInfo = this.serverViewInfo, Context = context });
+            }
+        }
+
+        public override Task Save(ServerViewContext context, ServerInfo serverInfo)
+        {
+            var ex = new NotSupportedException();
+            ex.Data["UserMessage"] += "ServerHandler does not support Save method";
+            throw ex;
+        }
+
+        public override Task<string> Script(ServerViewContext context, ServerInfo obj)
+        {
+            var ex = new NotSupportedException();
+            ex.Data["UserMessage"] += "ServerHandler does not support Script method";
+            throw ex;
+        }
+        public Task<Server?> CreateServerSmoObject(ServerViewContext context)
```

Comment:
Why is this public? And TBH, this doesn't even really seem necessary to create a whole new method for. Especially if you remove the GetSmoObject call.